File: presentation/src/main/java/vn/linh/tracker/feature/main/MainModule.java
package vn.linh.tracker.feature.main;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import vn.linh.tracker.infrastructure.RequestLocationProvider;
import vn.linh.tracker.infrastructure.RequestPermissionProvider;

@Module
public class MainModule {

    @Provides
    RequestPermissionProvider provideRequestPermissionHandler(MainActivity activity) {
        return new RequestPermissionProvider(activity);
    }

    @Provides
    RequestLocationProvider requestLocationProvider(MainActivity activity, Context context) {
        return new RequestLocationProvider(activity, context);
    }
}

File: presentation/src/main/java/vn/linh/tracker/feature/main/MainModule.kt
package vn.linh.tracker.feature.main

import android.content.Context

import dagger.Module
import dagger.Provides
import vn.linh.tracker.infrastructure.RequestLocationProvider
import vn.linh.tracker.infrastructure.RequestPermissionProvider

@Module(includes = [MainFragmentBuilderModule::class])
class MainModule {

    @Provides
    internal fun provideRequestPermissionHandler(activity: MainActivity): RequestPermissionProvider {
        return RequestPermissionProvider(activity)
    }

    @Provides
    internal fun provideRequestLocationProvider(activity: MainActivity, context: Context): RequestLocationProvider {
        return RequestLocationProvider(activity, context)
    }
}
